### PR TITLE
pull request (backend): Unit tests backend

### PR DESCRIPTION
**Pull request by Matisse-M**
This pull request introduces a comprehensive backend testing setup for the project, integrating Jest and Supertest for unit and integration tests, and updates the CI/CD pipeline to run backend tests before deployment. It also adds mock files for dependencies and improves error handling in backend routes. The most important changes are grouped below.

**Testing Infrastructure and Configuration**

* Added a Jest configuration file (`backend/jest.config.js`) to enable TypeScript ESM support, coverage collection, and custom test roots and setup.
* Updated backend dependencies in `backend/package.json` to include `jest`, `supertest`, `ts-jest`, and related type definitions, and added new test scripts (`test`, `test:watch`, `test:coverage`). [[1]](diffhunk://#diff-495707834ca4b862f9acdfbac70d279023d2c059da13db59594e61ed3354fed5L10-R12) [[2]](diffhunk://#diff-495707834ca4b862f9acdfbac70d279023d2c059da13db59594e61ed3354fed5R34-R48)
* Added mock files for database (`backend/tests/__mocks__/db.ts`), GitHub OAuth (`backend/tests/__mocks__/github.oauth.ts`), and JWT secret (`backend/tests/__mocks__/index.ts`) to support isolated testing. [[1]](diffhunk://#diff-4eda2d01cdd55a9d9b5f5fc738a4ea53031e6f22c3c5f70aeb0d37db0e51d0a2R1-R5) [[2]](diffhunk://#diff-424278fe84f24c90365450723d5f5ff8f659ffc61ea67c1e37348421771c4814R1-R6) [[3]](diffhunk://#diff-ea9f9d86bf2afab11e4add8233d441fdedcb55650d8be84481137b4734950718R1)

**Test Coverage for Backend Logic**

* Added extensive unit tests for about route utilities (`backend/tests/routes/about.utils.test.ts`), covering IP extraction, service mapping, timestamp conversion, error handling, and edge cases.
* Added integration tests for authentication routes (`backend/tests/routes/auth.test.ts`), verifying login, registration, logout, verification, password reset, and error responses.

**CI/CD Pipeline Enhancements**

* Updated `.github/workflows/cicd.yml` to add a `backend-test` job that runs backend tests after linting, and made deployment dependent on successful backend tests and web linter. [[1]](diffhunk://#diff-6727e33ccc9195d67f0786e1384f8f1cdaf4090c3e77547943105bd2b28c99d0R103-R123) [[2]](diffhunk://#diff-6727e33ccc9195d67f0786e1384f8f1cdaf4090c3e77547943105bd2b28c99d0L131-R153)

**Minor Improvements and Cleanup**

* Removed unnecessary console logging from the backend auth route for cleaner error handling.

### CODE REVIEW

File: backend/tests/routes/about.utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,348 @@
+// Test unitaires pour les fonctions utilitaires de about.ts
```
Comment (eliotttesnier) [inline backend/tests/routes/about.utils.test.ts:1]: Put comment in English or remove them

**Review by eliotttesnier — CHANGES_REQUESTED** [review]

**Review by eliotttesnier — APPROVED** [review]
lgtm
